nets/nets.go

```diff
@@ -0,0 +1,211 @@
+package nets
+
+import (
+	"errors"
+	"net"
+	"net/netip"
+	"time"
+)
+
+//go:linkname UseSocketStack net.useNetdev
+func UseSocketStack(stack SocketStack)
+
+// GethostByName() errors
+var (
+	ErrHostUnknown = errors.New("host unknown")
+	ErrMalAddr     = errors.New("malformed address")
+)
+
+// Socket errors
+var (
+	ErrFamilyNotSupported   = errors.New("address family not supported")
+	ErrProtocolNotSupported = errors.New("socket protocol/type not supported")
+	ErrStartingDHCPClient   = errors.New("error starting DHPC client")
+	ErrNoMoreSockets        = errors.New("no more sockets")
+	ErrClosingSocket        = errors.New("error closing socket")
+)
+
+var (
+	ErrConnected         = errors.New("already connected")
+	ErrConnectFailed     = errors.New("connect failed")
+	ErrConnectTimeout    = errors.New("connect timed out")
+	ErrMissingSSID       = errors.New("missing WiFi SSID")
+	ErrAuthFailure       = errors.New("wifi authentication failure")
+	ErrAuthTypeNoGood    = errors.New("wifi authorization type not supported")
+	ErrConnectModeNoGood = errors.New("connect mode not supported")
+	ErrNotSupported      = errors.New("not supported")
+)
+
+const (
+	_AF_INET       = 0x2
+	_SOCK_STREAM   = 0x1
+	_SOCK_DGRAM    = 0x2
+	_SOL_SOCKET    = 0x1
+	_SO_KEEPALIVE  = 0x9
+	_SOL_TCP       = 0x6
+	_TCP_KEEPINTVL = 0x5
+	_IPPROTO_TCP   = 0x6
+	_IPPROTO_UDP   = 0x11
+	// Made up, not a real IP protocol number.  This is used to create a
+	// TLS socket on the device, assuming the device supports mbed TLS.
+	_IPPROTO_TLS = 0xFE
+	_F_SETFL     = 0x4
+)
+
+// Interface is the minimum interface that need be implemented by any network
+// device driver and is based on [net.Interface].
+type Interface interface {
+	// HardwareAddr6 returns the device's 6-byte [MAC address].
+	//
+	// [MAC address]: https://en.wikipedia.org/wiki/MAC_address
+	HardwareAddr6() ([6]byte, error)
```

Comment:
net.HardwareAddr?

Reply:
I'd avoid net.HardwareAddr for the same reasons we switched to netip.Addr, it'd be a pain to have to start checking probe implementations for the implementation details of the HardwareAddr method and if the memory was copied or a reference was passed. Also there's the issue of heap memory usage. All in all, if we can avoid net.HardwareAddr I'd do so.